### PR TITLE
Fix bugs in pixel data collection

### LINKED ISSUE #332: Extension often doesn't collect the "origin_url" for pixels
The new extension version added pixel-capturing probes. These are supposed to also capture the webpage which the pixels are on, but that seems to not work in the majority of cases (based on looking at the data in the analysis environment).

Currently it uses `details.initiator`. That might need to change.

### LINKED ISSUE #333: TikTok pixel collection seems to be missing the actual parameters of the pixel data
We can see that a pixel was found, but the parameters seem to be lost (vs. Google/Meta pixels where we can see what data was sent via the pixel)

### PR DESCRIPTION
**Pull request by aaga**
Closes #332 
Closes #333

### CODE REVIEW

File: extensions/attention-stream/src/background.ts [hunk 1]
```diff
@@ -561,7 +579,7 @@ class GetPingsUploader extends Uploader {
       "youtube-video-recommendations": "id",
       "youtube-ads": "id",
       "tracking-pixel": "id",
-      "attribution": "id"
```
Comment (rhelmer) [inline extensions/attention-stream/src/background.ts:564]: Hm, why just this one? :)

File: extensions/attention-stream/src/background.ts [hunk 2]
```diff
@@ -409,31 +411,47 @@ async function stateChangeCallback(newState) {
...
+            // or raw request body submitted as well.
             let formData;
+            let rawRequestBody;
             if (details.method === "POST") {
               const rawFormData = details.requestBody?.formData;
+              const rawRequestData = details.requestBody?.raw;
               if (rawFormData) {
                 formData = new URLSearchParams(rawFormData).toString();
               }
+              if (rawRequestData) {
```
Comment (rhelmer) [inline extensions/attention-stream/src/background.ts:424]: Does this mean we will unconditionally collect both the parsed form and also the raw data, if the tracker uses a multipart/form?

Comment (rhelmer) [reply extensions/attention-stream/src/background.ts:424]: I can imagine a rationale for collecting both but I'm curious what your thoughts here are - it seems like if an already-parsed form is available we should prefer that, and only get the raw request data if it isn't.

Comment (aaga) [reply extensions/attention-stream/src/background.ts:424]: yeah I was erring on the side of "collect redundant data vs. potentially miss data". It looks like the spec is that if the data is parsed into form data then the raw is empty anyway?

File: extensions/attention-stream/src/background.ts (hunk 1 above)
Comment (aaga) [reply extensions/attention-stream/src/background.ts:564]: oh I think that's the default prettier setting (prefer non-quotes and only use quotes if needed i.e. keys with weird characters). I agree it looks kinda funny and would prefer "if one gets quotes all should get quotes" but I also don't want to be explicitly undoing the prettier format every time.

if we decide on the latter as our standard though, I can update my prettier settings?

File: extensions/attention-stream/src/background.ts (hunk 2 above)
Comment (rhelmer) [reply extensions/attention-stream/src/background.ts:424]: Oh interesting, does the raw body get consumed if there's a form? If so I guess it's a moot point.

**Review by rhelmer — APPROVED** [review]

**Pull request merged**